src/2_branch.js

```diff
@@ -139,182 +172,184 @@ Branch.prototype['createLink'] = function(obj, callback) {
 	this.api(resources.createLink, obj, function(err, data) {
 		if (typeof callback == 'function') {
 			if (err) { callback(err); }
-			else { callback(null, data['url']); }
+			else { callback(data['url']); }
 		}
 	});
 };
 
-
-/*
-===== THESE ARE OLD FUNCTIONS FROM THE WEB SDK THAT NEED TO BE MOVED OVER =====
-
-this.createLinkClick = function(url, callback) {
-	if (!self.initialized) { return utils.console(config.debugMsgs.nonInit); }
-	self.api.makeRequest(config.resources.links.createLinkClick, {
-		url: url
-	}, function(data) {
-		if (typeof callback == 'function') { callback(data.click_id); }
+/**
+ * @param {?String} url
+ * @param {?function} callback
+ */
+Branch.prototype['createLinkClick'] = function(url, callback) {
+	callback = callback || function() {};
+	if (!this.initialized) { return utils.console(utils.messages.nonInit); }
+	
+	this.api(resources.createLinkClick, {
+		link_url: url.replace('https://bnc.lt/', ''),
+		click: "click"
+	}, function(err, data) {
+		if (err) { callback(err); }
+		else { callback(null, data); }
 	});
 };
-this.SMSLink = function(obj, callback) {
-	if (!self.initialized) { return utils.console(config.debugMsgs.nonInit); }
-	var phone = obj.phone;
-	obj.channel = 'sms';
-	if (config.linkId === undefined) {
-		this.createLink(obj, function(url) {
-			self.api.makeRequest(config.resources.links.createLinkClick, {
-				link_url: url.replace(/^http:\/\/[^\/]+/, 'https://bnc.lt') + '?click'
-			}, function(data) {
-				self.sendSMSLink(phone, config.linkUrl + '/c/' + data.click_id, function() {
-					if (typeof callback == 'function') { callback(); }
+
+/**
+ * @param {?Object} metadata
+ * @param {?function} callback
+ */
+Branch.prototype['SMSLink'] = function(obj, callback) {
+	callback = callback || function() {};
+	if (!this.initialized) { return utils.console(utils.messages.nonInit); }
+	obj["channel"] = 'sms';
+	var self = this;
+	this.createLink(obj, function(url) {
```

Comment:
Well, the idea is, if the user clicked a link, use that, but users didn't always click a link, so if they didn't, just create a new one. But I could see also falling back to some link you pass in instead of creating a new one every time. Does that make sense?